### PR TITLE
Fix md_link_converter to not strip off .md suffix for external links

### PR DESCRIPTION
**Pull request by chessbyte**
Fixes https://github.com/ManageIQ/manageiq_docs/issues/555

### CODE REVIEW

File: lib/miq/md_link_converter.rb [hunk 1]
```diff
@@ -24,10 +26,21 @@ def with_prefixed_path
 
     private
 
+    def external_link?(url)
+      begin
+        uri = URI(url)
+      rescue
+        return false # in case caller does not pass in valid URL
+      end
+      return false if uri.scheme.nil? || uri.host.nil?
```
Comment (Fryguy) [inline lib/miq/md_link_converter.rb:35]: I think there's a `uri.relative?` that might be cleaner than checking manually.

File: lib/miq/md_link_converter.rb [hunk 2]
```diff
@@ -24,10 +26,21 @@ def with_prefixed_path
...
+      return false if uri.host.start_with?("manageiq.org") || uri.host.start_with?("www.manageiq.org")
+      return true
+    end
+
     def remove_ext(str)
       str.gsub(MD_LINK) do |link|
         title, path = link.match(MD_LINK)[1..2]
-        "[#{title}](#{path.gsub(".md", '')})"
+        external_link?(path) ? "[#{title}](#{path})" : "[#{title}](#{path.gsub(".md", '')})"
```
Comment (Fryguy) [inline lib/miq/md_link_converter.rb:43]:
```suggestion
        path.gsub!(".md", '') unless external_link?(path)
        "[#{title}](#{path})"
```

File: test/lib/miq/md_link_converter_test.rb [hunk 3]
```diff
@@ -25,4 +25,11 @@ def test_multiline_conversion
 
     assert_equal expected, Miq::MdLinkConverter.new(original).convert
   end
+
+  def test_converting_external_md_href
+    original = "[Title](https://github.com/ManageIQ/ui-components/blob/master/README.md)"
+    expected = "[Title](https://github.com/ManageIQ/ui-components/blob/master/README.md)"
+
+    assert_equal expected, Miq::MdLinkConverter.new(original).convert
+  end
```
Comment (Fryguy) [inline test/lib/miq/md_link_converter_test.rb:34]: Can you add tests for the manageiq.org cases?

File: lib/miq/md_link_converter.rb (hunk 1 above)
Comment (chessbyte) [reply lib/miq/md_link_converter.rb:35]: `uri.relative?` is the same as `uri.scheme.nil?`.  Would still need to check `uri.host.nil?`, I believe

File: lib/miq/md_link_converter.rb (hunk 2 above)
Comment (chessbyte) [reply lib/miq/md_link_converter.rb]: addressed

File: test/lib/miq/md_link_converter_test.rb (hunk 3 above)
Comment (chessbyte) [reply test/lib/miq/md_link_converter_test.rb:34]: addressed

**Pull request merged**